src/main/kotlin/no/nav/syfo/soknad/SoknadService.kt

```diff
@@ -23,11 +23,31 @@ class SoknadService(
     }
 
     fun handleSykepengesoknad(sykepengesoknad: SykepengesoknadDTO) {
-        if ((sykepengesoknad.status == SoknadsstatusDTO.SENDT && sykepengesoknad.sendtArbeidsgiver != null) &&
-            sykepengesoknad.tom?.isAfter(LocalDate.now().minusMonths(4)) == true
-        ) {
-            soknadDb.insert(sykepengesoknad.toSoknadDbModel())
+        if (shouldHandleSoknad(sykepengesoknad)) {
+            when (sykepengesoknad.status) {
+                SoknadsstatusDTO.NY -> soknadDb.insertOrUpdate(sykepengesoknad.toSoknadDbModel())
+                SoknadsstatusDTO.FREMTIDIG -> soknadDb.insertOrUpdate(sykepengesoknad.toSoknadDbModel())
+                SoknadsstatusDTO.SENDT -> handleSendt(sykepengesoknad)
+                SoknadsstatusDTO.KORRIGERT -> handleSendt(sykepengesoknad)
+                SoknadsstatusDTO.AVBRUTT -> soknadDb.deleteSoknad(sykepengesoknad.id)
+                SoknadsstatusDTO.SLETTET -> soknadDb.deleteSoknad(sykepengesoknad.id)
+            }
         }
         SOKNAD_TOPIC_COUNTER.inc()
     }
+
+    private fun handleSendt(sykepengesoknad: SykepengesoknadDTO) {
+        when (sykepengesoknad.sendtArbeidsgiver != null) {
+            true -> soknadDb.insertOrUpdate(sykepengesoknad.toSoknadDbModel())
+            else -> soknadDb.deleteSoknad(sykepengesoknad.id)
```

Comment:
Er dette for å gjøre noe med de korrigerte..? Mener Håvard sa at korrigerte søknader alltid vil komme som en ny søknadsid, og at en søknad aldri vil endre seg når den først er sendt, men sikkert verdt å sjekke på nytt.

Reply:
Tenker det her blir riktig, fordi hvis den ble kun sendt til nav så må den forsvinne herfra